src/tpm2/VendorString.h

```diff
@@ -62,47 +62,67 @@
 
-/*     The following #if macro may be deleted after a proper VENDOR_STRING_1 is provided. */
-#ifndef VENDOR_STRING_1
-#error VENDOR_STRING_1 is not provided.					\
-    Please modify VendorString.h to provide a vendor specific string.
+/*      To customize the MANUFACTURER_ID macro, use the configure option "--with-manufacturer-id=XXXX"
+        Define up to a 4-byte hex value for MANUFACTURER_ID.  This value defines the response for
+        manufacturer in TPMAttributes for TPM2_GetInfo().
+        Must be in the TPM Vendor ID Registry: https://trustedcomputinggroup.org/resource/vendor-id-registry/ */
+#ifndef MANUFACTURER_ID
+#define MANUFACTURER_ID    0x53494d30
 #endif
+_Static_assert((sizeof(MANUFACTURER_ID) <= 4U), "MANUFACTURER_ID can be up to 4-bytes");
 
-/* the more significant 32-bits of a vendor-specific value indicating the version of the firmware
-   The following line should be un-commented and a vendor specific firmware V1 should be provided
-   here. The FIRMWARE_V2 may also be defined as appropriate. */
-#define   FIRMWARE_V1         (0x20191023)
+/*      To customize the VENDOR_STRING_[1-4] macros, use the configure options "--with-vendor-string-[1-4]=XXXX"
+        Define up to 4, 4-byte, vendor-specific strings. The strings must each be 4 bytes long.
+        These values define the response for TPM_PT_VENDOR_STRING_(1-4) in TPM2_GetCapability(). */
+#ifndef CONFIG_VENDOR_STRING_1
+#define VENDOR_STRING_1       "fTPM"
+#else
+#define VENDOR_STRING_1 STRINGIFY(CONFIG_VENDOR_STRING_1)
+#endif
+_Static_assert(sizeof(VENDOR_STRING_1) - 1U == 4U, "VENDOR_STRING_1 must be 4-bytes");
 
-// the less significant 32-bits of a vendor-specific value indicating the version of the firmware
-#define   FIRMWARE_V2         (0x00163636)
+#ifdef CONFIG_VENDOR_STRING_2
+#define VENDOR_STRING_2 STRINGIFY(CONFIG_VENDOR_STRING_2)
+_Static_assert(sizeof(VENDOR_STRING_2) - 1U == 4U, "VENDOR_STRING_2 must be 4-bytes");
+#endif
+#ifdef CONFIG_VENDOR_STRING_3
+#define VENDOR_STRING_3 STRINGIFY(CONFIG_VENDOR_STRING_3)
+_Static_assert(sizeof(VENDOR_STRING_3) - 1U == 4U, "VENDOR_STRING_3 must be 4-bytes");
+#endif
+#ifdef CONFIG_VENDOR_STRING_4
+#define VENDOR_STRING_4 STRINGIFY(CONFIG_VENDOR_STRING_4)
+_Static_assert(sizeof(VENDOR_STRING_4) - 1U == 4U, "VENDOR_STRING_4 must be 4-bytes");
+#endif
 
-// The following #if macro may be deleted after a proper FIRMWARE_V1 is provided.
+/*      To customize the FIRMWARE_V[1-2] macros, use the configure options "--with-firmware-v[1-2]=XXXX"
+        Define the more significant 32-bits of a vendor-specific value indicating the
+        version of the firmware in FIRMWARE_V1. The less significant 32-bits of a vendor-specific
+        value indicating the version of the firmware can use the FIRMWARE_V2 macro. */
 #ifndef FIRMWARE_V1
-#error  FIRMWARE_V1 is not provided.					\
-    Please modify VendorString.h to provide a vendor specific firmware \
-    version
+#define FIRMWARE_V1     0x00000000
```

Comment:
Also this here should go back to its previous default value. None of the existing users, like distros, should have to modify their build process and `./configure` line to get back the old default values. So **all** values have to default to what they were before.